### PR TITLE
ArmourStands and Entities

### PR DESCRIPTION
**Pull request by AzureusNation**
All entities have the option to be invisible or not so I have moved the getter and setter to Entity. Also I have added noGravity to ArmorStand as that was missing when I was implementing.

Only thing with Entities is that they may have their own setInvisible method so in the implementations the setInvisible will need to be Override so that the correct minecraft method can be ran.

### CODE REVIEW

File: src/main/java/org/spongepowered/api/entity/living/ArmorStand.java [hunk 1]
```diff
@@ -132,18 +132,18 @@
     void setSmall(boolean small);
 
     /**
-     * Returns whether this armor stand is rendered invisible.
+     * Returns whether this armor stand is affected by gravity or not.
      *
-     * @return Whether this armor stand is invisible
+     * @return Whether this is a small armor stand
```
Comment (kenzierocks) [inline src/main/java/org/spongepowered/api/entity/living/ArmorStand.java]: This seems incorrect, is this supposed to be a different method?

File: src/main/java/org/spongepowered/api/entity/living/ArmorStand.java [hunk 2]
```diff
@@ -132,18 +132,18 @@
...
      *
-     * @return Whether this armor stand is invisible
+     * @return Whether this is affected by gravity or not
      */
-    boolean isInvisible();
+    boolean hasNoGravity();
 
     /**
-     * Sets whether this armor stand is invisible or not.
+     * Returns whether this armor stand is affected by gravity or not.
```
Comment (Minecrell) [inline src/main/java/org/spongepowered/api/entity/living/ArmorStand.java]: The previous Javadocs were correct, this sets something, but doesn't return something.

File: src/main/java/org/spongepowered/api/entity/living/ArmorStand.java [hunk 3]
```diff
@@ -132,18 +132,18 @@
     void setSmall(boolean small);
 
     /**
-     * Returns whether this armor stand is rendered invisible.
+     * Returns whether this armor stand is affected by gravity or not.
      *
-     * @return Whether this armor stand is invisible
+     * @return Whether this is affected by gravity or not
```
Comment (Minecrell) [inline src/main/java/org/spongepowered/api/entity/living/ArmorStand.java]: `this armor stand is ...` sounds better than `this is ...`, only `this` would refer to the operation of the method, and not armor stand.

**Comment by Minecrell** [comment]
Wouldn't `hasGravity` and `setGravity` make more sense than an additional negation with `setNoGravity`? While the property of the armor stand is defined as `NoGravity`, the method would make more sense the other way around, IMO

**Comment by AzureusNation** [comment]
> Wouldn't  hasGravity  and  setGravity  make more sense than an additional negation with  setNoGravity ? While the property of the armor stand is defined as  NoGravity , the method would make more sense the other way around, IMO

@Minecrell I can change that, no problem

**Pull request closed without merging**